### PR TITLE
fix PULSEDEV-N/A Dockerfile: Fix image build by updating CA certificates

### PR DESCRIPTION
**Pull request by ghost**
It was no longer working because certificates from old ubuntu version were no longer valid.

### CODE REVIEW

**Comment by AlbertoEAF** [comment]
The comment is not entirely accurate yet, I suggest changing to:

This image uses very specific (and old) versions of the OS and tools. The aim is two-fold. First, to match the build pipeline used for the official Microsoft LightGBM  releases, for which there are official tests, avoiding unexpected bugs. Second, retain libc==2.12, to ensure compatibility of the produced LightGBM binaries with any target machine.

**Review by AlbertoEAF — APPROVED** [review]

File: docker/lightgbm-ci-build-env/Dockerfile [hunk 1]
```diff
@@ -60,8 +64,22 @@ RUN apt-get update \
...
     wget \
     zip \
+    make \
+    tar \
  && rm -rf /var/lib/apt/lists/*
 
+# Update ca certificates to safely perform curls
+RUN mkdir -p ~/.ca-certificates-src \
+    && cd ~/.ca-certificates-src \
+    && wget https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/ca-certificates/20210119~20.04.2/ca-certificates_20210119~20.04.2.tar.xz \
```
Comment (shengwangsw) [inline docker/lightgbm-ci-build-env/Dockerfile]: for security reason, consider downloading the file.
the red flag is that you are executing `sudo make and sudo make install`

Comment (ghost) [reply docker/lightgbm-ci-build-env/Dockerfile]: Great point. fixed

**Review by AlbertoEAF — APPROVED** [review]

**Review by shengwangsw — APPROVED** [review]
LGTM

**Pull request merged**